### PR TITLE
replace pad with pad-def

### PR DESCRIPTION
**Pull request by tjknoth**
Replace pad with pad-def as part of deprecation process (jitx-client PR 4759)

### CODE REVIEW

File: tests/landpatterns/BGA/pads.stanza [hunk 1]
```diff
@@ -34,7 +34,7 @@ deftest(BGA) test-bga-pad:
   val row = padrows[0]
   val lp-pad = row[0]
 
-  val p = pad(lp-pad)
+  val p = /pad-def(lp-pad)
```
[the remark this answers is not in the document]
Comment (tjknoth) [reply tests/landpatterns/BGA/pads.stanza:37]: It needs to be qualified to the root scope because there's a locally-bound `val pad-def = ...` above

**Review by bhusang — APPROVED** [review]

File: tests/landpatterns/BGA/pads.stanza (hunk 1 above)
[the remark this answers is not in the document]
Comment (callendorph) [reply tests/landpatterns/BGA/pads.stanza:37]: Let's rename that local variable

**Comment by callendorph** [comment]
What release is the new `pad-def` API function going to be included in?
This needs to wait until that release to merge. 
Alternatively - we need to introduce a `main` vs `develop` similar to `jitx-client`

**Comment by tjknoth** [comment]
> What release is the new `pad-def` API function going to be included in? This needs to wait until that release to merge. Alternatively - we need to introduce a `main` vs `develop` similar to `jitx-client`

That jitx-client PR has been stuck with a CI issue for a while -- I'll try to get it through today but it might not make it until 3.17

**Review by callendorph — APPROVED** [review]
I fixed a few trailing instances of `pad()` and now I don't see any warnings in the test log. 
:+1:

**Pull request merged**